### PR TITLE
feat: docker pull before run

### LINKED ISSUE #374: Install otomi from a chart
# User story
As a user, I want to install otomi on a cluster with a helm chart, so I can access the console immediately.

This makes it easier for others to install otomi.

## Use cases

1. A user can install otomi from a remote chart (providing a pull secret)
2. A user can start a proxy to the console

## Definition Of Ready

- [ ] Requirements have been reviewed, agreed and locked in discussion: 
- [ ] The story is self-contained and has no dependencies
- [ ] The story is reviewed and approved by the stake holders
- [ ] The story has been estimated in complexity points
- [ ] The story use cases are clear and can be translated into test scenarios
- [ ] Atomic SMART tasks have been created that reflect the functional requirements from the discussion
- [ ] SMART tasks reflect any dependencies on other tasks in this story (affecting order of delivery)

## Tasks

core:
1. [ ] Create `charts/install` that holds `templates/job.yaml` to run `bin/install.sh` script that installs: gitea, api and console and shows the isntructions to access it afterwards:
- start proxy command: `kubectl -n shared port-forward svc/otomi-console 8080:80`
- show url to acces the proxied service: `http://127.0.0.1:8080`

## Definition Of Done

- [ ] [Tasks](##Tasks) are done.
- [ ] (Unit) Tests are added with the code (if relevant and practical)
- [ ] (Architecture) Design Record(s) have been added as `adr/*.md` and appended to list in `adr/_index.md`.
- [ ] Schema and demo files have been updated to reflect code changes (if relevant).
- [ ] Documentation has been updated (if relevant).
- [ ] Functionality, code, and/or documentation have been peer reviewed.

### PR DESCRIPTION
**Pull request by Morriz**
fixes #374

### CODE REVIEW

**Comment by Dunky13** [comment]
Why is TIME changed to TRACE in this PR? Is it relevant to the docker pull? Rest looks good.

File: bin/otomi [hunk 1]
```diff
@@ -93,7 +92,7 @@ function show_usage() {
   Env flags:
     CI=1                Exit the script after first error
     DEBUG=1             Run otomi CLI in debug mode
-    TIME=1              Time otomi CLI calls
+    TRACE=1              Time otomi CLI calls
```
Comment (j-zimnowoda) [inline bin/otomi:95]: Copy paste issue. Please provide correct description.

File: bin/otomi [hunk 2]
```diff
@@ -211,6 +210,7 @@ function drun() {
   # - in docker AND force docker
   if [ "$has_docker" = 'true' ] && [ -z "$IN_DOCKER" ]; then
     [ -n "$VERBOSE" ] && echo "Running dockerized version of command: $command"
+    [ "$image_tag" = 'latest' ] || [ "$image_tag" = 'master' ] && [ "$has_docker" = 'true' ] && docker pull "otomi/core:$image_tag"
```
Comment (j-zimnowoda) [inline bin/otomi:213]: Thank you!

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Thanks for that change. A minor change is needed.

File: bin/otomi (hunk 1 above)
Comment (Morriz) [reply bin/otomi]: done

Comment (j-zimnowoda) [reply bin/otomi]: Thank you!

**Review by j-zimnowoda — APPROVED** [review]

**Comment by j-zimnowoda** [comment]
@Morriz, please do not use [ci skip] because build-test-push-release as `GitHub Expected — Waiting for status to be reported`

File: bin/otomi (hunk 2 above)
Comment (svatwork) [reply bin/otomi:214]: I see it checks for $has_docker two times? (line 212 & 214).

File: bin/common.sh [hunk 3]
```diff
@@ -65,8 +65,9 @@ function get_k8s_version() {
 }
 
 function otomi_image_tag() {
-  local otomiVersion=$([ -n "${CLOUD+x}${CLUSTER+x}" ] && yq r $clusters_file "clouds.$CLOUD.clusters.$CLUSTER.otomiVersion")
-  [ -n "$otomiVersion" ] && echo $otomiVersion || echo 'latest'
+  local otomi_version=$([ -n "${CLOUD+x}${CLUSTER+x}" ] && yq r $clusters_file "clouds.$CLOUD.clusters.$CLUSTER.otomiVersion")
+  [ -n "$otomi_version" ] && otomi_version='latest'
```
Comment (svatwork) [inline bin/common.sh:69]: This means: if `$otomi_version` is non-zero length `otomi_version='latest'`. That is not correct...

File: bin/otomi (hunk 2 above)
Comment (Morriz) [reply bin/otomi:214]: you may refactor it in your next delivery ;)

File: bin/common.sh (hunk 3 above)
Comment (Morriz) [reply bin/common.sh:69]: is there no test for that? can you add that?

File: bin/otomi (hunk 2 above)
Comment (svatwork) [reply bin/otomi:214]: #392

File: bin/common.sh (hunk 3 above)
Comment (svatwork) [reply bin/common.sh:69]: #392

**Pull request merged**